indico/modules/events/abstracts/forms.py

```diff
@@ -161,3 +173,54 @@ class CreateEmailTemplateForm(EditEmailTemplateRuleForm):
         ('reject', _('Reject')),
         ('merge', _('Merge'))
     ], description=_("The template that will be used as a basis for this notification. You can customize it later."))
+
+
+class AbstractForm(IndicoForm):
+    title = StringField(_("Title"), [DataRequired()])
+    description = IndicoMarkdownField(_('Content'), [DataRequired()], editor=True, mathjax=True)
+    submitted_contrib_type = QuerySelectField(_("Type"), get_label='name', allow_blank=True,
+                                              blank_text=_("No type selected"))
+    person_links = AbstractPersonLinkListField(_("People"), [DataRequired()])
+    submission_comment = TextAreaField(_("Comments"))
+    attachments = FileField(_('Attachments'), param_name='attachments', lightweight=True)
+
+    def __init__(self, *args, **kwargs):
+        self.event = kwargs.pop('event')
+        self.abstract = kwargs.pop('abstract', None)
+        super(AbstractForm, self).__init__(*args, **kwargs)
+        self.submitted_contrib_type.query = self.event.contribution_types
+        if not self.submitted_contrib_type.query.count():
+            del self.submitted_contrib_type
+        if abstracts_settings.get(self.event, 'allow_attachments'):
+            tpl = get_template_module('forms/_dropzone_themes.html')
+            self.attachments.widget_options['previewTemplate'] = tpl.thin_preview_template()
```

Comment:
This shouldn't be done here but in the `FieldField` based on the `lightweight` option